proposals/0033-data-collections.md

```diff
@@ -0,0 +1,244 @@
+---
+
+<h1>Our Authors</h1>
+<ul>
+  {authors.map(author => (
+    <li><a href={author.data.twitter}>{author.data.name}</a></li>
+  ))}
+</ul>
+```
+
+To retrieve individual entries, `getEntry()` can be used. This receives both the collection name and the entry `id` as described in the [Return type](#return-type). These can be passed as separate arguments or as object keys.
+
+```astro
+---
+// src/pages/en/index.astro
+
+// Ex. retrieve a translations document stored in `src/content/i18n/en.json`
+// Option 1: separate args
+const english = await getEntry('i18n', 'en');
+// Option 2: object keys
+const english = await getEntry({ collection: 'i18n', id: 'en' });
+---
+
+<h1>{english.data.homePage.title}</h1>
+<p>{english.data.homePage.tagline}</p>
+```
+
+> Note: the object keys approach is primarily meant for resolving references. See the [collection references RFC](https://github.com/withastro/roadmap/blob/d89e2a4c28379108501aa6bf40d2f8d93d81ad02/proposals/0034-collection-references.md) for more.
+
+Thanks to the generic function name, this can be used as a replacement for `getEntryBySlug()` as well. When querying a content collection, `getEntry()` uses `slug` as the identifier:
+
+
+```astro
+---
+// Option 1: separate args
+const welcomePost = await getEntry('blog', 'welcome');
+// Option 2: object keys
+const welcomePost = await getEntry({ collection: 'blog', slug: 'welcome' });
+const Content = await welcomePost.render();
+---
+
+<h1>{welcomePost.data.title}</h1>
+<Content />
+```
+
+### `id` vs `slug`
+
+You may have noticed two competing identifiers depending on the collection type: `id` for data, and `slug` for content. This is an inconsistency we'd like to address in a future RFC, with `id` becoming the new standard for identifying collection entries. For now, `slug` will remain on content collections to make the introduction of data collections non-breaking.
+
+**Full background:** `slug` was originally introduced alongside the content `id` to have a URL-friendly version of the file path, which can be passed to `getStaticPaths()` for route generation. Data collections are not intended to be used as routes, so we don't want to perpetuate this pattern. `slug` also "slugifies" the file path by removing capitalization and replacing spaces with dashes. If we added this processing to data collection IDs, [collection references](https://github.com/withastro/roadmap/blob/d89e2a4c28379108501aa6bf40d2f8d93d81ad02/proposals/0034-collection-references.md) will be less intuitive to use (i.e. "do I include spaces in the referenced ID here?").
+
+## Implementation
+
+Data collections should following the API design of content collections with a stripped-down featureset. To wire up data collections, we will introduce an internal utility that mirrors our `addContentEntryType()` integration function. This example registers a new `dataEntryType` for `.json` files, with necessary logic to parse data as a JS object:
+
+```ts
+addDataEntryType({
```

Comment:
Just confirming that `JSON` support is built-in to Astro without needing the user to define `addDataEntryType()` themselves, correct? Are any other data types supported out of the box (csv? yaml?) It would be good to make a note of this here in the RFC if this is purely illustrative and not expected that every user copy-pastes this snippet into their project.

Reply:
Ah yes, I should definitely clarify! JSON is built-in, and this API is only detailed to show how the feature is implemented internally. Nothing else is included at the moment but I think YAML is worth adding before release; this would make copying frontmatter into a data file more direct.

Reply:
JSON + YAML seems like reasonable defaults to me.

Reply:
+1 for YAML